core/Nonce.php

```diff
@@ -127,28 +127,26 @@ public static function getOrigin()
     public static function getAcceptableOrigins()
     {
         $host = Url::getCurrentHost(null);
-        $port = '';
-
-        // parse host:port
-        if (preg_match('/^([^:]+):([0-9]+)$/D', $host, $matches)) {
-            $host = $matches[1];
-            $port = $matches[2];
-        }
 
         if (empty($host)) {
             return array();
         }
 
-        // standard ports
-        $origins = array(
-            'http://' . $host,
-            'https://' . $host,
-        );
-
-        // non-standard ports
-        if (!empty($port) && $port != 80 && $port != 443) {
-            $origins[] = 'http://' . $host . ':' . $port;
-            $origins[] = 'https://' . $host . ':' . $port;
+        // parse host:port
+        if (preg_match('/^([^:]+):([0-9]+)$/D', $host, $matches)) {
+            $host = $matches[1];
+            $port = $matches[2];
+            $origins = array(
+                'http://' . $host . $port,
+                'https://' . $host . $port,
+            );
+        } else {
+            $origins = array(
+                'http://' . $host,
+                'https://' . $host,
+                'http://' . $host . ':80',
```

Comment:
Thanks for the PR @luismsgomes 

btw always adding these ports not sure if it's a good idea or has any side effects. Wonder if we could remove the  two lines that always add `:80` and `:443` assuming that your `$host` is like `https://somedomain.org:443`

What is `Url::getCurrentHost(null);` returning for you?

Reply:
Hi,

The main motivation for this PR was exactly that `:80` and `:443` were missing from the list of urls.
The url returned by `Url::getCurrentHost(null);` contains `https://` and `:443` in my case.

Cheers.

Reply:
@luismsgomes added few suggestions to prevent we're regressing here on anything.

Basically could even use the logic completely but only change the `if (!empty($port) && $port != 80 && $port != 443) {` to `if (!empty($port)) {`

Basically.... if no port is configured, maybe we don't want to add `80` and `443`. I think this be still fine for your case. It might be fine to add it just not sure and while it's not needed be good to maybe not add it automatically

Reply:
Sorry, my last reply was incorrect:  getCurrentHost() returns only the hostname, no scheme and no port in my case.
The function getOrigin() does return the scheme and the port.

Reply:
I see 👍 I'll undo my suggestions. @Findus23 I reckon it be fine to also allow the port as we can assume if no port is returned it is running on either port 80 or 443? Maybe if SSL is forced (`[General]force_ssl=1`), then we would not return the URL with port `80`?

Reply:
> Maybe if SSL is forced ([General]force_ssl=1), then we would not return the URL with port 80?

Yes, that makes sense.

Reply:
@luismsgomes  can you make that change or do you need help with this?

Reply:
Done.